src/test/java/com/monov/student/controller/StudentControllerIntTest.java

```diff
@@ -0,0 +1,100 @@
+package com.monov.student.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.monov.commons.dto.ItemIds;
+import com.monov.student.StudentServiceApplication;
+import com.monov.student.entity.Student;
+import com.monov.student.service.StudentService;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+        webEnvironment = SpringBootTest.WebEnvironment.MOCK,
+        classes = StudentServiceApplication.class)
+@AutoConfigureMockMvc
+@TestPropertySource(properties = { "spring.config.location = classpath:integration-test-application.yaml" })
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```

Comment:
Why do the tests have a fixed order? Test A creates data that is used by test B?

Reply:
order removed. Data changes to DB are rolled back after each test